### ISSUE TITLE
Error on `make build` when compiling celestia node from source

### ISSUE THREAD

**Issue by claudioantonio** [celestiaorg/celestia-node#3215]
### Celestia Node version

b560d25e103d891d33f8e268e5c1f1e9e8dced84

### OS

Ubuntu 22.04 on top of Windows 11 with wsl 2

### Install tools

Makefile

### Steps to reproduce it

- git clone https://github.com/celestiaorg/celestia-node.git
- cd celestia-node
- make build

### Expected result

Build successfully

### Actual result

# github.com/pyroscope-io/godeltaprof/internal/pprof
../../../go/pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.2/internal/pprof/delta_mutex.go:25:20: undefined: runtime_cyclesPerSecond
../../../go/pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.2/internal/pprof/proto.go:400:8: undefined: runtime_expandFinalInlineFrame
make: *** [Makefile:38: build] Error 1

### Relevant log output

```Shell
claudio@LAPTOP-MJVHICCL:~/src/cartesi/celestia-node$ make build
--> Building Celestia
go: downloading github.com/cristalhq/jwt v1.2.0
go: downloading github.com/ipfs/go-log/v2 v2.5.1
go: downloading github.com/filecoin-project/go-jsonrpc v0.3.1
go: downloading github.com/spf13/cobra v1.8.0
go: downloading go.opentelemetry.io/otel v1.22.0
go: downloading go.opentelemetry.io/otel/metric v1.22.0
go: downloading go.uber.org/fx v1.20.1
go: downloading github.com/celestiaorg/celestia-app v1.4.0
go: downloading github.com/ipfs/go-log v1.0.5
go: downloading github.com/celestiaorg/cosmos-sdk v1.18.3-sdk-v0.46.14
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/pyroscope-io/otel-profiling-go v0.5.0
go: downloading go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp v0.45.0
go: downloading go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp v1.22.0
go: downloading cosmossdk.io/math v1.2.0
go: downloading github.com/libp2p/go-libp2p v0.32.2
go: downloading go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.22.0
go: downloading github.com/multiformats/go-multiaddr v0.12.2
go: downloading github.com/multiformats/go-base32 v0.1.0
go: downloading go.uber.org/dig v1.17.1
go: downloading go.uber.org/multierr v1.11.0
go: downloading golang.org/x/sys v0.17.0
go: downloading golang.org/x/xerrors v0.0.0-20231012003039-104605ab7028
go: downloading github.com/BurntSushi/toml v1.3.2
go: downloading github.com/celestiaorg/go-fraud v0.2.0
go: downloading github.com/celestiaorg/go-header v0.5.3
go: downloading github.com/dgraph-io/badger/v4 v4.2.1-0.20240106094458-1c417aa3799c
go: downloading github.com/imdario/mergo v0.3.16
go: downloading github.com/ipfs/boxo v0.18.0
go: downloading github.com/ipfs/go-datastore v0.6.0
go: downloading github.com/ipfs/go-ds-badger4 v0.1.5
go: downloading github.com/libp2p/go-libp2p-pubsub v0.10.0
go: downloading github.com/stretchr/testify v1.8.4
go: downloading github.com/pyroscope-io/client v0.7.2
go: downloading go.opentelemetry.io/contrib/instrumentation/runtime v0.45.0
go: downloading go.opentelemetry.io/otel/sdk/metric v1.22.0
go: downloading go.opentelemetry.io/otel/sdk v1.22.0
go: downloading go.opentelemetry.io/otel/trace v1.22.0
go: downloading go.uber.org/zap v1.26.0
go: downloading github.com/ipfs/go-blockservice v0.5.0
go: downloading github.com/libp2p/go-libp2p-kad-dht v0.25.2
go: downloading github.com/libp2p/go-libp2p-routing-helpers v0.7.3
go: downloading github.com/multiformats/go-multiaddr-dns v0.3.1
go: downloading github.com/prometheus/client_golang v1.18.0
go: downloading golang.org/x/crypto v0.19.0
go: downloading github.com/celestiaorg/nmt v0.20.0
go: downloading github.com/mattn/go-isatty v0.0.20
go: downloading github.com/celestiaorg/rsmt2d v0.11.0
go: downloading github.com/go-logr/logr v1.4.1
go: downloading cosmossdk.io/errors v1.0.1
go: downloading github.com/cosmos/ibc-go/v6 v6.2.0
go: downloading github.com/spf13/cast v1.5.0
go: downloading github.com/celestiaorg/celestia-core v1.29.0-tm-v0.34.29
go: downloading github.com/tendermint/tm-db v0.6.7
go: downloading github.com/cosmos/cosmos-sdk/api v0.1.0
go: downloading google.golang.org/grpc v1.60.1
go: downloading go.opentelemetry.io/proto/otlp v1.0.0
go: downloading google.golang.org/protobuf v1.32.0
go: downloading golang.org/x/exp v0.0.0-20240110193028-0dcbfd608b1e
go: downloading github.com/libp2p/go-flow-metrics v0.1.0
go: downloading github.com/ipfs/go-cid v0.4.1
go: downloading github.com/mr-tron/base58 v1.2.0
go: downloading github.com/multiformats/go-multicodec v0.9.0
go: downloading github.com/multiformats/go-multihash v0.2.3
go: downloading github.com/multiformats/go-multistream v0.5.0
go: downloading github.com/multiformats/go-multibase v0.2.0
go: downloading github.com/multiformats/go-varint v0.0.7
go: downloading github.com/rollkit/go-da v0.4.0
go: downloading github.com/google/uuid v1.5.0
go: downloading go.opencensus.io v0.24.0
go: downloading github.com/cosmos/gogoproto v1.4.11
go: downloading github.com/regen-network/protobuf v1.3.3-alpha.regen.1
go: downloading github.com/grpc-ecosystem/grpc-gateway v1.16.0
go: downloading github.com/golang/protobuf v1.5.3
go: downloading google.golang.org/genproto/googleapis/api v0.0.0-20240108191215-35c7eff3a6b1
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/celestiaorg/dagstore v0.0.0-20230824094345-537c012aa403
go: downloading github.com/gorilla/mux v1.8.1
go: downloading github.com/hashicorp/go-retryablehttp v0.7.5
go: downloading golang.org/x/sync v0.6.0
go: downloading google.golang.org/genproto v0.0.0-20240102182953-50ed04b92917
go: downloading github.com/ipfs/go-block-format v0.2.0
go: downloading github.com/ipfs/go-ipld-format v0.6.0
go: downloading github.com/libp2p/go-buffer-pool v0.1.0
go: downloading github.com/libp2p/go-msgio v0.3.0
go: downloading github.com/jbenet/goprocess v0.1.4
go: downloading github.com/pyroscope-io/godeltaprof v0.1.2
go: downloading github.com/ipfs/go-ipld-cbor v0.1.0
go: downloading github.com/ipld/go-car v0.6.2
go: downloading github.com/gammazero/workerpool v1.1.3
go: downloading github.com/celestiaorg/go-libp2p-messenger v0.2.0
go: downloading github.com/ipfs/go-ipfs-delay v0.0.1
go: downloading github.com/ipfs/go-metrics-interface v0.0.1
go: downloading github.com/hashicorp/golang-lru/v2 v2.0.7
go: downloading github.com/hashicorp/golang-lru v1.0.2
go: downloading github.com/ipfs/bbloom v0.0.4
go: downloading github.com/quic-go/quic-go v0.40.1
go: downloading github.com/decred/dcrd/dcrec/secp256k1/v4 v4.2.0
go: downloading github.com/libp2p/go-netroute v0.2.1
go: downloading github.com/hashicorp/golang-lru/arc/v2 v2.0.5
go: downloading github.com/pbnjay/memory v0.0.0-20210728143218-7b4eea64cf58
go: downloading github.com/benbjohnson/clock v1.3.5
go: downloading github.com/raulk/go-watchdog v1.3.0
go: downloading github.com/google/gopacket v1.1.19
go: downloading github.com/libp2p/go-libp2p-kbucket v0.6.3
go: downloading github.com/ipfs/go-verifcid v0.0.2
go: downloading github.com/ipfs/go-ipfs-exchange-interface v0.2.1
go: downloading github.com/ipfs/go-ipfs-blockstore v1.3.1
go: downloading github.com/libp2p/go-libp2p-record v0.2.0
go: downloading github.com/whyrusleeping/go-keyspace v0.0.0-20160322163242-5b898ac5add1
go: downloading gonum.org/v1/gonum v0.14.0
go: downloading github.com/miekg/dns v1.1.57
go: downloading github.com/Jorropo/jsync v1.0.1
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/cespare/xxhash/v2 v2.2.0
go: downloading github.com/cespare/xxhash v1.1.0
go: downloading github.com/prometheus/client_model v0.5.0
go: downloading github.com/prometheus/common v0.45.0
go: downloading github.com/prometheus/procfs v0.12.0
go: downloading github.com/go-logr/stdr v1.2.2
go: downloading github.com/cosmos/cosmos-proto v1.0.0-alpha8
go: downloading github.com/celestiaorg/quantum-gravity-bridge/v2 v2.1.2
go: downloading github.com/celestiaorg/merkletree v0.0.0-20210714075610-a84dc3ddbbe4
go: downloading github.com/klauspost/reedsolomon v1.11.8
go: downloading github.com/99designs/keyring v1.2.1
go: downloading github.com/cosmos/go-bip39 v1.0.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/btcsuite/btcd/btcec/v2 v2.3.2
go: downloading github.com/tendermint/go-amino v0.16.0
go: downloading sigs.k8s.io/yaml v1.3.0
go: downloading github.com/gogo/gateway v1.1.0
go: downloading github.com/gorilla/handlers v1.5.1
go: downloading github.com/spf13/viper v1.14.0
go: downloading github.com/confio/ics23/go v0.9.1
go: downloading github.com/armon/go-metrics v0.4.1
go: downloading github.com/grpc-ecosystem/go-grpc-middleware v1.3.0
go: downloading github.com/regen-network/cosmos-proto v0.3.1
go: downloading golang.org/x/net v0.20.0
go: downloading github.com/go-kit/log v0.2.1
go: downloading github.com/go-logfmt/logfmt v0.5.1
go: downloading github.com/cenkalti/backoff/v4 v4.2.1
go: downloading github.com/grpc-ecosystem/grpc-gateway/v2 v2.19.0
go: downloading github.com/google/btree v1.1.2
go: downloading github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20240108191215-35c7eff3a6b1
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: downloading github.com/multiformats/go-base36 v0.2.0
go: downloading github.com/hashicorp/go-cleanhttp v0.5.2
go: downloading github.com/ipld/go-car/v2 v2.13.1
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/ipfs/go-ipfs-util v0.0.3
go: downloading github.com/google/go-cmp v0.6.0
go: downloading github.com/dgraph-io/ristretto v0.1.1
go: downloading github.com/dustin/go-humanize v1.0.1
go: downloading github.com/ipfs/go-ipld-legacy v0.2.1
go: downloading github.com/ipld/go-codec-dagpb v1.6.0
go: downloading github.com/ipld/go-ipld-prime v0.21.0
go: downloading github.com/cskr/pubsub v1.0.2
go: downloading github.com/gammazero/deque v0.2.0
go: downloading github.com/ipfs/go-peertaskqueue v0.8.1
go: downloading github.com/polydawn/refmt v0.89.0
go: downloading github.com/whyrusleeping/cbor-gen v0.0.0-20240109153615-66e95c3e8a87
go: downloading github.com/multiformats/go-multiaddr-fmt v0.1.0
go: downloading github.com/libp2p/go-yamux/v4 v4.0.1
go: downloading github.com/jbenet/go-temp-err-catcher v0.1.0
go: downloading github.com/flynn/noise v1.0.1
go: downloading github.com/libp2p/go-libp2p-asn-util v0.4.1
go: downloading github.com/klauspost/compress v1.17.4
go: downloading github.com/libp2p/go-reuseport v0.4.0
go: downloading github.com/marten-seemann/tcp v0.0.0-20210406111302-dfbc87cc63fd
go: downloading github.com/mikioh/tcpinfo v0.0.0-20190314235526-30a79bb1804b
go: downloading github.com/quic-go/webtransport-go v0.6.0
go: downloading github.com/ipfs/go-merkledag v0.11.0
go: downloading github.com/libp2p/go-nat v0.2.0
go: downloading github.com/opentracing/opentracing-go v1.2.0
go: downloading github.com/containerd/cgroups v1.1.0
go: downloading github.com/elastic/gosigar v0.14.2
go: downloading github.com/minio/sha256-simd v1.0.1
go: downloading github.com/libp2p/go-cidranger v1.1.0
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/ipfs/go-ipfs-ds-help v1.1.1
go: downloading go.uber.org/atomic v1.11.0
go: downloading github.com/matttproud/golang_protobuf_extensions/v2 v2.0.0
go: downloading github.com/bgentry/speakeasy v0.1.0
go: downloading github.com/dvsekhvalnov/jose2go v1.5.0
go: downloading github.com/godbus/dbus v0.0.0-20190726142602-4481cbc300e2
go: downloading github.com/gsterjov/go-libsecret v0.0.0-20161001094733-a6f4afe4910c
go: downloading github.com/mtibben/percent v0.2.1
go: downloading golang.org/x/term v0.17.0
go: downloading github.com/rakyll/statik v0.1.7
go: downloading github.com/klauspost/cpuid/v2 v2.2.6
go: downloading github.com/mitchellh/mapstructure v1.5.0
go: downloading github.com/spf13/afero v1.9.2
go: downloading github.com/spf13/jwalterweatherman v1.1.0
go: downloading github.com/cosmos/btcutil v1.0.5
go: downloading github.com/cosmos/iavl v0.19.6
go: downloading github.com/go-kit/kit v0.12.0
go: downloading github.com/manifoldco/promptui v0.9.0
go: downloading github.com/felixge/httpsnoop v1.0.1
go: downloading github.com/rcrowley/go-metrics v0.0.0-20201227073835-cf1acfcdf475
go: downloading github.com/hashicorp/go-immutable-radix v1.3.1
go: downloading github.com/rs/zerolog v1.31.0
go: downloading github.com/improbable-eng/grpc-web v0.15.0
go: downloading github.com/cometbft/cometbft-db v0.7.0
go: downloading github.com/lib/pq v1.10.7
go: downloading github.com/rs/cors v1.8.2
go: downloading go.opentelemetry.io/otel/exporters/stdout/stdouttrace v1.21.0
go: downloading github.com/google/flatbuffers v1.12.1
go: downloading github.com/petar/GoLLRB v0.0.0-20210522233825-ae3b015fd3e9
go: downloading github.com/golang/glog v1.1.2
go: downloading github.com/whyrusleeping/cbor v0.0.0-20171005072247-63513f603b11
go: downloading github.com/ipfs/go-ipfs-pq v0.0.3
go: downloading github.com/davidlazar/go-crypto v0.0.0-20200604182044-b73af7476f6c
go: downloading github.com/francoispqt/gojay v1.2.13
go: downloading github.com/quic-go/qpack v0.4.0
go: downloading github.com/mikioh/tcpopt v0.0.0-20190314235656-172688c1accc
go: downloading github.com/huin/goupnp v1.3.0
go: downloading github.com/jackpal/go-nat-pmp v1.0.2
go: downloading github.com/koron/go-ssdp v0.0.4
go: downloading github.com/coreos/go-systemd/v22 v22.5.0
go: downloading github.com/docker/go-units v0.5.0
go: downloading github.com/godbus/dbus/v5 v5.1.0
go: downloading github.com/opencontainers/runtime-spec v1.1.0
go: downloading github.com/hdevalence/ed25519consensus v0.0.0-20220222234857-c00d1f31bab3
go: downloading github.com/ChainSafe/go-schnorrkel v1.0.0
go: downloading github.com/subosito/gotenv v1.4.1
go: downloading gopkg.in/ini.v1 v1.67.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/magiconair/properties v1.8.6
go: downloading github.com/pelletier/go-toml/v2 v2.0.7
go: downloading golang.org/x/text v0.14.0
go: downloading github.com/gtank/merlin v0.1.1
go: downloading github.com/chzyer/readline v1.5.1
go: downloading github.com/golang/snappy v0.0.5-0.20220116011046-fa5810519dcb
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/pelletier/go-toml v1.9.5
go: downloading lukechampine.com/blake3 v1.2.1
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/coinbase/rosetta-sdk-go v0.7.9
go: downloading github.com/Workiva/go-datastructures v1.0.53
go: downloading github.com/creachadair/taskgroup v0.3.2
go: downloading github.com/influxdata/influxdb-client-go/v2 v2.12.2
go: downloading github.com/google/orderedcode v0.0.1
go: downloading github.com/mattn/go-colorable v0.1.13
go: downloading github.com/desertbit/timer v0.0.0-20180107155436-c41aec40b27f
go: downloading nhooyr.io/websocket v1.8.7
go: downloading github.com/tidwall/btree v1.5.0
go: downloading filippo.io/edwards25519 v1.0.0-rc.1
go: downloading github.com/gtank/ristretto255 v0.1.2
go: downloading github.com/mimoo/StrobeGo v0.0.0-20210601165009-122bf33a46e0
go: downloading github.com/consensys/gnark-crypto v0.12.1
go: downloading github.com/shirou/gopsutil v3.21.6+incompatible
go: downloading github.com/influxdata/line-protocol v0.0.0-20210311194329-9aa0e372d097
go: downloading github.com/cockroachdb/apd/v2 v2.0.2
go: downloading github.com/deepmap/oapi-codegen v1.8.2
go: downloading github.com/hashicorp/go-getter v1.7.0
go: downloading github.com/aws/aws-sdk-go v1.44.122
go: downloading cloud.google.com/go/storage v1.30.1
go: downloading cloud.google.com/go v0.111.0
go: downloading github.com/bgentry/go-netrc v0.0.0-20140422174119-9fd32a8b3d3d
go: downloading github.com/hashicorp/go-safetemp v1.0.0
go: downloading github.com/hashicorp/go-version v1.6.0
go: downloading github.com/mitchellh/go-testing-interface v1.14.1
go: downloading github.com/ulikunitz/xz v0.5.10
go: downloading golang.org/x/oauth2 v0.16.0
go: downloading google.golang.org/api v0.149.0
go: downloading cloud.google.com/go/compute/metadata v0.2.3
go: downloading cloud.google.com/go/iam v1.1.5
go: downloading github.com/googleapis/gax-go/v2 v2.12.0
go: downloading cloud.google.com/go/compute v1.23.3
go: downloading github.com/jmespath/go-jmespath v0.4.0
go: downloading github.com/google/s2a-go v0.1.7
go: downloading github.com/googleapis/enterprise-certificate-proxy v0.3.2
go: downloading github.com/bits-and-blooms/bitset v1.7.0
# github.com/pyroscope-io/godeltaprof/internal/pprof
../../../go/pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.2/internal/pprof/delta_mutex.go:25:20: undefined: runtime_cyclesPerSecond
../../../go/pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.2/internal/pprof/proto.go:400:8: undefined: runtime_expandFinalInlineFrame
make: *** [Makefile:38: build] Error 1
```

### Notes

go version go1.22.0 linux/amd64

**Comment by ramin** [comment]
@claudioantonio ack

is this ONLY when using WSL? I'll see if i can replicate

**Comment by ramin** [comment]
@claudioantonio just noticed, this is related to https://github.com/celestiaorg/celestia-node/issues/3178 which we should fix this week, if you can use go 1.21 for now, that'll work for you, i'll ping you here once we push fix

**Comment by claudioantonio** [comment]
Thanks a lot @ramin !
